packages/app-admin-ui/client/providers/MasterList.js

```diff
@@ -0,0 +1,174 @@
+/* global KEYSTONE_ADMIN_META */
+
+import React, { useContext, createContext } from 'react';
+
+import { useQuery } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+
+import List from '../classes/List';
+import FetchingListsPage from '../pages/FetchingLists';
+import { useAdminMeta } from './AdminMeta';
+
+import { views, readViews, preloadViews } from '../FIELD_TYPES';
+
+// TODO: eventually, we'll get everything through the query below and won't need this
+const { lists: adminMetaListData } = KEYSTONE_ADMIN_META;
+
+const MasterListContext = createContext();
+
+const LISTS_QUERY = gql`
+  query getAllLists {
+    meta: _ksListsMeta(where: { auxiliary: false }) {
+      key
+      path
+      label
+      singular
+      plural
+      description
+      access {
+        create
+        read
+        update
+        delete
+      }
+      schema {
+        sortType
+        queries {
+          item
+          list
+          meta
+        }
+        mutations {
+          create
+          createMany
+          update
+          updateMany
+          delete
+          deleteMany
+        }
+        inputTypes {
+          whereInput
+          createInput
+          createManyInput
+          updateInput
+          updateManyInput
+        }
+      }
+    }
+  }
+`;
+
+/**
+ * Handles metadata on all available lists.
+ */
+export const MasterListProvider = ({ children }) => {
```

Comment:
should we use a better name for this? `ListMetaProvider`? and `useListmeta` may be?

Reply:
I like. 👍